src/ui/actions/timeline.ts

```diff
@@ -467,65 +469,93 @@ export function setFocusRegion(focusRegion: FocusRegion | null): UIThunkAction {
       dispatch(stopPlayback());
     }
 
-    if (focusRegion !== null) {
-      const zoomRegion = getZoomRegion(state);
-      const { endTime: prevEndTime, startTime: prevStartTime } = getFocusRegion(state) || {};
-
-      let { endTime, startTime } = focusRegion;
+    if (focusRegion === null) {
```

Comment:
:speech_balloon: not suggesting this has to be done now, but most of this calculation _really_ feels like something that belongs in a reducer instead if at all possible.

Reply:
Yes, absolutely.  That's part of the point of Redux.  We _want_ as much logic as possible in a reducer, because pure functions are easier to test and predict.

We also specifically recommend dispatching fewer actions, and modeling actions as "events" rather than "setters":

- https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers
- https://redux.js.org/style-guide/#model-actions-as-events-not-setters
- https://redux.js.org/style-guide/#avoid-dispatching-many-actions-sequentially
- https://redux.js.org/style-guide/#allow-many-reducers-to-respond-to-the-same-action